ui/src/configuration/components/Variables.tsx

```diff
@@ -0,0 +1,200 @@
+// Libraries
+import React, {PureComponent, ChangeEvent} from 'react'
+import _ from 'lodash'
+import {connect} from 'react-redux'
+
+// Utils
+import {
+  getVariables,
+  createVariable,
+  updateVariable,
+  deleteVariable,
+} from 'src/variables/actions'
+
+// Components
+import TabbedPageHeader from 'src/shared/components/tabbed_page/TabbedPageHeader'
+import CreateVariableOverlay from 'src/configuration/components/CreateVariableOverlay'
+import {Button, ComponentSize, RemoteDataState} from '@influxdata/clockface'
+import VariableList from 'src/organizations/components/VariableList'
+import {Input, Overlay, EmptyState} from 'src/clockface'
+import FilterList from 'src/shared/components/Filter'
+import AddResourceDropdown from 'src/shared/components/AddResourceDropdown'
+
+// Types
+import {ComponentColor, IconFont} from '@influxdata/clockface'
+import {OverlayState} from 'src/types'
+import {AppState} from 'src/types/v2'
+import {Variable, Organization} from '@influxdata/influx'
+import {VariablesState} from 'src/variables/reducers'
+
+interface StateProps {
+  variables: VariablesState
+  orgs: Organization[]
+}
+
+interface DispatchProps {
+  onCreateVariable: typeof createVariable
+  onUpdateVariable: typeof updateVariable
+  onDeleteVariable: typeof deleteVariable
+}
+
+type Props = StateProps & DispatchProps
+
+interface State {
+  searchTerm: string
+  createOverlayState: OverlayState
+  importOverlayState: OverlayState
+  variables: Variable[]
+}
+
+class Variables extends PureComponent<Props, State> {
+  public state: State = {
+    searchTerm: '',
+    createOverlayState: OverlayState.Closed,
+    importOverlayState: OverlayState.Closed,
+    variables: this.variableList(this.props.variables),
+  }
+
+  public render() {
+    const {orgs, variables} = this.props
+    const {searchTerm, createOverlayState} = this.state
+
+    return (
+      <>
+        <TabbedPageHeader>
+          <Input
+            icon={IconFont.Search}
+            placeholder="Filter variables..."
+            widthPixels={290}
+            value={searchTerm}
+            onChange={this.handleFilterChange}
+            onBlur={this.handleFilterBlur}
+          />
+          <AddResourceDropdown
+            resourceName="Variable"
+            onSelectImport={this.handleOpenImportOverlay}
+            onSelectNew={this.handleOpenCreateOverlay}
+          />
+        </TabbedPageHeader>
+        <FilterList<Variable>
+          searchTerm={searchTerm}
+          searchKeys={['name']}
+          list={this.variableList(variables)}
+          sortByKey="name"
+        >
+          {v => (
+            <VariableList
+              variables={v}
+              emptyState={this.emptyState}
+              onDeleteVariable={this.handleDeleteVariable}
+              onUpdateVariable={this.handleUpdateVariable}
+            />
+          )}
+        </FilterList>
+        <Overlay visible={createOverlayState === OverlayState.Open}>
```

Comment:
Lets move this overlay component into the CreateVariableOverlay and have that take a visible prop; this will make it easier to move this into the router.

Reply:
Got it, I added it as the prop